src/AllocatorRegistry.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+contract AllocatorRegistry {
+    // --- storage variables ---
+
+    mapping(address => uint256) public wards;
+    mapping(bytes32 => address) public buffers;
+
+    // --- events ---
+
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed ilk, bytes32 indexed what, address data);
+
+    // --- modifiers ---
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "AllocatorRegistry/not-authorized");
+        _;
+    }
+
+    // --- constructor ---
+
+    constructor() {
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    // --- administration ---
+
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    function file(bytes32 ilk, bytes32 what, address data) external auth {
+        if (what == "buffer") {
+            buffers[ilk] = data;
```

Comment:
note: multiple ilks can map to the same buffer; I'm assuming this isn't an issue and is maybe even an intended feature, but wanted to flag it just in case

Reply:
Yes, this isn't an issue as the pause proxy is not expected to do that (map multiple ilks to the same buffer).